canvas_sdk/client/base.py

```diff
@@ -105,24 +109,33 @@ def call(action, url, request_context, params=None, data=None, max_retries=None,
         try:
             # build and send the request
             response = canvas_session.request(
-                action, url, params=params, data=data, headers=headers, cookies=cookies,
-                files=files, auth=auth, timeout=timeout, proxies=proxies, verify=verify,
-                cert=cert, allow_redirects=allow_redirects)
+                action, url, params=params, data=data, headers=headers,
+                cookies=cookies, files=files, auth=auth, timeout=timeout,
+                proxies=proxies, verify=verify, cert=cert,
+                allow_redirects=allow_redirects)
 
             # raise an http exception if one occured
             response.raise_for_status()
 
         except HTTPError as http_error:
+            log.info("Caught an API Error returned by Canvas: %s", str(http_error))
             # Need to check its an error code that can be retried
-            status_code = http_error.response.status_code
+            status_code = response.status_code
```

Comment:
This used to use `http_error` because it was worried about the actual `canvas_session.request()` call triggering the exception.  If it does, `response` will be `None`, and this exception handler will blow up, masking the original error.

We can fix that by either putting a simpler exception handler around *just* the `canvas_session.request()` call, or by putting the error handling that requires `response` to exist inside an if clause.  I'd probably argue for the latter, but could be talked into the former.